### DISCUSSION TITLE
Datagridview sorting

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3762] [Questions] [answered]
I am sure this is very simple : )
We have a Winform application where users asked for sorting on columns. Since the datagridview is fed by 'BusinessListBase' I thought sorting would be available by default. I tried LinkObservable collection as well but no luck.

Regards

**Comment by rockfordlhotka** [comment]
Windows Forms requires that you use `BusinessBindingListBase`. That is the collection type that supports the old WinForms data binding interfaces.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  Thank you Sir. I will change the collection.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka I changed the collection to BusinessBindingListBase, couldn't get the automatic sorting to work so I tried Programmatic sorting but it me gave this error
"DataGridView control cannot be sorted if it is bound to an IBindingList that does not support sorting.
  Source=System.Windows.Forms
  StackTrace:
   at System.Windows.Forms.DataGridView.SortDataBoundDataGridView_PerformCheck(DataGridViewColumn dataGridViewColumn)
   at System.Windows.Forms.DataGridView.SortInternal(IComparer comparer, DataGridViewColumn dataGridViewColumn, ListSortDirection direction)
"

Kind Regards

**Reply by Chicagoan2016** [reply, marked as answer]
@rockfordlhotka I recalled from almost twenty years ago we have SortedBindingList in Csla.
I tried it and it's working : )
Is this the correct solution?

Thank you

**Reply by swegele** [reply]
FYI - we use Telerik Grids and sorting/filtering just works, we don't make any changes at all to code.

**Reply by Chicagoan2016** [reply]
@swegele I think I have to try Telerik Grid now. If I bind to SortedBindingList, a checkbox column in datagridview is not binding to  property of the underlying object (child of collection).
